File: 1899-count-items-matching-a-rule/solution.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int colIdx;
        switch (ruleKey) {
            case "type":
                colIdx = 0;
                break;
            case "color":
                colIdx = 1;
                break;
            default:
                colIdx = 2;
                break;
        }

        int count = 0;
        for (List<String> list : items) {
            if (list.get(colIdx).equals(ruleValue)) {
                count++;
            }
        }
        return count;
    }
}

File: 1899-count-items-matching-a-rule/solution.go
func countMatches(items [][]string, ruleKey string, ruleValue string) int {
    count := 0
    ruleIdx := 0
    if ruleKey == "color" {
        ruleIdx = 1
    } else if ruleKey == "name" {
        ruleIdx = 2
    }

    for _, item := range items {
        if item[ruleIdx] == ruleValue {
            count++
        }
    }

    return count
}
